fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraBaseResource.java

```diff
@@ -71,28 +71,17 @@ protected HttpIdentifierConverter identifierConverter() {
         return identifierConverter;
     }
 
-    protected FedoraResource getFedoraResource(final FedoraId fedoraId) throws PathNotFoundException {
-        return this.resourceFactory.getResource(fedoraId);
-    }
-
     /**
      * Gets a fedora resource by id. Uses the provided transaction if it is uncommitted,
      * or uses a new transaction.
      *
      * @param transaction the fedora transaction
-     * @param fedoraId identifier of the resource
+     * @param fedoraId    identifier of the resource
      * @return the requested FedoraResource
      */
-    protected FedoraResource getFedoraResource(final Transaction transaction, final FedoraId fedoraId) {
-        try {
-            if (transaction.isCommitted()) {
-                return getFedoraResource(fedoraId);
-            } else {
-                return resourceFactory.getResource(transaction, fedoraId);
-            }
-        } catch (final PathNotFoundException e) {
-            throw new PathNotFoundRuntimeException(e);
-        }
+    protected FedoraResource getFedoraResource(final Transaction transaction, final FedoraId fedoraId)
+            throws PathNotFoundException {
+            return resourceFactory.getResource(TransactionUtils.openTxId(transaction), fedoraId);
```

Comment:
It seems like every call to `ResourceFactory.getResource()` outside of `org.fcrepo.kernel.impl.models` goes through `TransactionUtils.openTxId(transaction)` or an equivalent implementation. 

Would it make sense to add back in a `ResourceFactory.getResource(Transaction tx, FedoraId resourceId)` signature which does `TransactionUtils.openTxId` internally and calls the `getResource(String txId, FedoraId resourceId)` rather than calling `TransactionUtils.openTxId` everywhere?

Reply:
Yes if not for the fact that IndexBuilderImpl does not have access to a Transaction but rather interacts directly with the  fcrepo-persistence-ocfl layer directly.  Consequently DbSearchIndexImpl doesn't want to deal in Transaction Objects, but does need to know about the transaction in which the rebuild is taking place.  @pwinckles  was in favor of removing the dependency of the Transaction from the ResourceFactory API since it encumbers the interface without providing a commensurate benefit.

Reply:
@bbpennel ^^

Reply:
Oh - I think I misunderstood you.

Reply:
You're saying simply add the other signature and use where appropriate yes?

Reply:
Right, there'd be multiple `getResource` signatures, one would take a Transaction, another would take a txId. Other parts of the application would call them where appropriate and then getResource would internally do the null and committed check, since that seems to be a requirement everywhere that a transaction object is used. `doesResourceExist` could do the same thing if that's valuable, but I didn't investigate that one closely. The benefit would be that a cleaner usage and we'd be less likely to forget that when using a transaction we have to do those checks on it since it'd be centralized.

Reply:
That makes sense to me